packages/xstate-react/src/useMachine.ts

```diff
@@ -114,141 +96,86 @@ export function useMachine<
   Interpreter<TContext, any, TEvent, TTypestate>['send'],
   Interpreter<TContext, any, TEvent, TTypestate>
 ] {
-  const machine = useConstant(() => {
-    return typeof getMachine === 'function' ? getMachine() : getMachine;
-  });
-
-  if (
-    process.env.NODE_ENV !== 'production' &&
-    typeof getMachine !== 'function'
-  ) {
-    const [initialMachine] = useState(machine);
-
-    if (machine !== initialMachine) {
-      console.warn(
-        'Machine given to `useMachine` has changed between renders. This is not supported and might lead to unexpected results.\n' +
-          'Please make sure that you pass the same Machine as argument each time.'
-      );
-    }
-  }
-
-  const {
-    context,
-    guards,
-    actions,
-    activities,
-    services,
-    delays,
-    state: rehydratedState,
-    ...interpreterOptions
-  } = options;
-
-  const [resolvedMachine, service] = useConstant<
-    [
-      StateNode<TContext, any, TEvent, TTypestate>,
-      Interpreter<TContext, any, TEvent, TTypestate>
-    ]
-  >(() => {
-    const machineConfig = {
-      context,
-      guards,
-      actions,
-      activities,
-      services,
-      delays
-    };
-    const machineWithConfig = machine.withConfig(machineConfig, {
-      ...machine.context,
-      ...context
-    } as TContext);
+  const service = useInterpret(getMachine, options);
 
-    return [
-      machineWithConfig,
-      interpret(machineWithConfig, { deferEvents: true, ...interpreterOptions })
-    ];
+  useEffect(() => {
```

Comment:
Reposting from https://github.com/davidkpiano/xstate/pull/1855#discussion_r559147349

The subscription has been happening in the layout effect previously - not anymore though. After more thinking about this in recent months I'm not totally sure which one is best, but I'm inclined to say that probably - for our use case - the layout effect is better. At least for the time being.

Reply:
Define "better"?